api/src/game_state_server.py

```diff
@@ -252,7 +299,15 @@ def _delete_token(self, token_id: str, room_id: str) -> None:
             del self._rooms[room_id].id_to_positions[token_id]
         # Remove the token from the state
         if self._rooms[room_id].game_state.get(token_id, False):
-            del self._rooms[room_id].game_state[token_id]
+            removed_token = self._rooms[room_id].pop_token(token_id)
+            # Remove token from icon_id table
+            if self._rooms[room_id].tokens_by_icon_id.get(removed_token.icon_id, False):
```

Comment:
you're doing self._rooms[room_id].tokens_by_icon_id.get(removed_token.icon_id) twice here. I think it would be clearer to extract that into a variable.